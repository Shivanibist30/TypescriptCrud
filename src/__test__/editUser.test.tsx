import Enzyme,{shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Edituser from '../components/editUsers/editUser';
import {toMatchImageSnapshot} from 'jest-image-snapshot';
import puppeteer from 'puppeteer-core';

jest.mock('../components/editUsers/editUser',()=>jest.fn());
const mockFn=jest.fn();
expect.extend({toMatchImageSnapshot});
Enzyme.configure({adapter:new Adapter()});
const browserPath="C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe";

describe('testing edit user component',()=>{
    it("Rendering Edituser component",()=>{
        const wrapper=shallow(<Edituser/>)
        console.log(wrapper)  
    });
    it("checking for div",()=>{
        const wrapper=shallow(<Edituser/>)
        expect(wrapper.contains('div'))
    });

    it("snapshot testing",()=>{
        const wrapper=shallow(<button name='update' onClick={mockFn}/>);
        wrapper.simulate('click')
        expect(mockFn).toMatchSnapshot()
    });

    // it('cancel event',()=>{
    //     const mockCallback=jest.fn();
    //     const wrapper=shallow(<Edituser/>)
    //     wrapper.find('.cancelButton').simulate('click')
        
    //     expect(mockCallback).toHaveBeenCalled()

    // });

    jest.setTimeout(300000)
    it('create edit user snapshot',async()=>{
    const browser = await puppeteer.launch({executablePath:browserPath,});
    const page=await browser.newPage();
    await page.goto('http://localhost:3000/editUser/');
    const image= await page.screenshot()
    expect(image).toMatchImageSnapshot()
    });
});