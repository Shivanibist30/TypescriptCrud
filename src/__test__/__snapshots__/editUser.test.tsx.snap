// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing edit user component snapshot testing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
